app/src/main/java/com/example/tasky/agenda/presentation/viewmodel/AgendaViewModel.kt

```diff
@@ -26,33 +28,120 @@ class AgendaViewModel @Inject constructor(
     private val agendaRepository: AgendaRepository,
 ) : ViewModel() {
 
+    private val selectedDate: MutableStateFlow<LocalDate> = MutableStateFlow(LocalDate.now())
+
     var state by mutableStateOf(AgendaState())
         private set
 
     init {
         state = state.copy(
-            days = dateGenerator.getWeek(),
             userInitials = GetInitialsUseCase(userCache.getUser()?.fullName),
-            selectedMonth = dateGenerator.getMonth()
         )
-        val now = LocalDate.now()
         viewModelScope.launch {
-            agendaRepository.getAgenda(now).collectLatest {
+            selectedDate.collect {
+                getAgendaForDate(it, shouldFetch = true)
                 state = state.copy(
-                    agendaItems = AddNeedleToAgenda(now.toEpochDay(), it.toMutableList())
+                    days = dateGenerator.getWeek(it),
+                    selectedMonth = dateGenerator.getMonth(it)
                 )
             }
         }
+    }
+
+    private fun getAgendaForDate(date: LocalDate, shouldFetch: Boolean = false) {
         viewModelScope.launch {
-            val fetchSuccessfully = agendaRepository.fetchAgenda(now)
+            agendaRepository.getAgenda(date).collect {
```

Comment:
Same as my previous comment


The bug mentioned on whatsapp, can happen due to an issue here

Which is, everytime you change the date, you are running a new viewmodelscope with a .collect in it

The collect, runs infinitely. Meaning you'd be running infinite coroutines here (one for each time the user changes a date)

Solution can be to cancel the previous job (viewmodelscope.launch) before creating a new one

Reply:
Yest it worked! I was wondering if that collect would be the problem, but didnt know how to solve it. Thanks a lot